client/my-sites/plugins/plugins-list/plugins-list-dataviews.tsx

```diff
@@ -0,0 +1,82 @@
+import { filterSortAndPaginate } from '@wordpress/dataviews';
+import { useTranslate } from 'i18n-calypso';
+import { useEffect, useMemo, useState } from 'react';
+import { initialDataViewsState } from 'calypso/a8c-for-agencies/components/items-dashboard/constants';
+import ItemsDataViews from 'calypso/a8c-for-agencies/components/items-dashboard/items-dataviews';
+import { DataViewsState } from 'calypso/a8c-for-agencies/components/items-dashboard/items-dataviews/interfaces';
+import { useSelector } from 'calypso/state';
+import { Plugin } from 'calypso/state/plugins/installed/types';
+import { getPluginActionStatuses } from '../plugin-management-v2/utils/get-plugin-action-statuses';
+import { useActions } from './use-actions';
+import { useFields } from './use-fields';
+
+import './style.scss';
+
+interface Props {
+	currentPlugins: Array< Plugin >;
+	initialSearch?: string;
+	isLoading: boolean;
+	onSearch?: ( search: string ) => void;
+	bulkActionDialog: ( action: string, plugins: Array< Plugin > ) => void;
+}
+
+export default function PluginsListDataViews( {
+	currentPlugins,
+	initialSearch,
+	isLoading,
+	onSearch,
+	bulkActionDialog,
+}: Props ) {
+	const allStatuses = useSelector( ( state ) => {
+		return getPluginActionStatuses( state );
+	} );
+
+	const translate = useTranslate();
+
+	const fields = useFields( bulkActionDialog );
+	const actions = useActions( bulkActionDialog );
+
+	const [ dataViewsState, setDataViewsState ] = useState< DataViewsState >( () => ( {
+		...initialDataViewsState,
+		perPage: 15,
+		search: initialSearch,
+		fields: [ 'plugins', 'sites', 'update' ],
+	} ) );
+
+	// When search changes, notify the parent component
+	useEffect( () => {
+		onSearch && onSearch( dataViewsState.search || '' );
+	}, [ dataViewsState.search, onSearch ] );
+
+	const { data, paginationInfo } = useMemo( () => {
+		const result = filterSortAndPaginate( currentPlugins, dataViewsState, fields );
+
+		// Add all statuses to the plugin object so we can display them in the DataViews
+		result.data.forEach( ( plugin ) => {
+			plugin.allStatuses = allStatuses.filter( ( status ) => status.pluginId === plugin.id );
+		} );
+
+		return {
+			data: result.data,
+			paginationInfo: result.paginationInfo,
+		};
+	}, [ currentPlugins, dataViewsState, fields, allStatuses ] );
+
+	return (
+		<ItemsDataViews
```

Comment:
What does this abstraction bring? In other words, why aren't we using `DataViews` directly?

Reply:
Agree. I think we should aim to remove `ItemsDataViews`, so the less we depend on it now the easier will be removing it down the road.

Reply:
That's a good point, no need to use it.
I used it because I saw more usage than direct `DataViews` on Calypso code. [Fixed it here](https://github.com/Automattic/wp-calypso/pull/95561/commits/8547dfac0a99aa25e9db3233656fceb672dcfeea).